### PR TITLE
feat(#1857): Use `FtCached` with correct `CacheVersion`

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
1. Add plugin version to the cache
2. Refactor old tests little bit
2.1. Reorder methods in `FakeMaven`
2.2. Remove redundant `public` modifiers

Closes: #1857

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@mximp Could you have a look?

**Comment by volodya-lombrozo** [comment]
@mximp Reminder

**Comment by volodya-lombrozo** [comment]
@mximp Reminder

File: eo-maven-plugin/src/main/java/org/eolang/maven/AssembleMojo.java [hunk 1]
```diff
@@ -235,6 +236,14 @@ public final class AssembleMojo extends SafeMojo {
     @SuppressWarnings("PMD.ImmutableField")
     private boolean unrollExitError = true;
 
+    /**
+     * The current version of eo-maven-plugin.
+     * Maven 3 only.
+     * @checkstyle MemberNameCheck (7 lines)
+     */
+    @Parameter(defaultValue = "${plugin}", readonly = true)
```
[the remark this answers is not in the document]
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/AssembleMojo.java:247]: @mximp I've added more informative description to the JavaDoc. Could you take a look again, please?

[the remark this answers is not in the document]
Comment (mximp) [reply eo-maven-plugin/src/main/java/org/eolang/maven/AssembleMojo.java:247]: Got it: `PluginParameterExpressionEvaluator` will convert the string.

**Review by mximp — APPROVED** [review]

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you merge that changes, please?

**Comment by volodya-lombrozo** [comment]
@yegor256 Reminder

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**